loader/src/main/java/net/neoforged/fml/loading/ClasspathTransformerDiscoverer.java

```diff
@@ -59,4 +70,38 @@ private void locateTransformers(String resource) throws IOException {
             found.add(new NamedPath(path.toUri().toString(), path));
         }
     }
+
+    private void scanModClasses() {
+        final Map<String, List<Path>> modClassPaths = getModClasses();
+        modClassPaths.forEach((modid, paths) -> {
+            if (shouldLoadInServiceLayer(paths)) {
+                found.add(new NamedPath(modid, paths.toArray(Path[]::new)));
+            }
+        });
+    }
+
+    private Map<String, List<Path>> getModClasses() {
```

Comment:
There is no need to duplicate this, I think we can make the method in `CommonLaunchHandler` static and call it directly.